### PR TITLE
Additions to support onos-config devices

### PR DESCRIPTION
**Pull request by shadansari**

### CODE REVIEW

File: api/topo/topo.go [hunk 1]
```diff
@@ -24,6 +30,25 @@ type ID string
...
+const (
+	Revision    = "revision"
+	Address     = "address"
+	Target      = "target"
+	Type        = "type"
+	Version     = "version"
+	Timeout     = "timeout"
+	Role        = "role"
+	Displayname = "displayname"
+	User        = "user"
```
Comment (adibrastegarnia) [inline api/topo/topo.go:43]: I am not sure we should we include these kind of information in device attributes like username and password? Perhaps I am understanding is wrong.

Comment (shadansari) [reply api/topo/topo.go:43]: @adibrastegarnia - I agree. I would also like to make onos-topo super generic but the fact of the matter is that users of onos-topo (specifically onos-config) expect this information. At this point, I thought this is the best compromise solution.

Comment (adibrastegarnia) [reply api/topo/topo.go:43]: @shadansari 
OK. makes sense.

Comment (SeanCondon) [reply api/topo/topo.go:43]: I think `revision` is not necessary for `onos-config` to have

File: api/topo/topo.go [hunk 2]
```diff
@@ -33,3 +58,86 @@ var TopoClientFactory = func(cc *grpc.ClientConn) TopoClient {
...
+		ID:        device.ID(obj.ID),
+		Protocols: []*device.ProtocolState{},
+	}
+
+	r, _ := strconv.Atoi(obj.Attributes[Revision])
+	d.Revision = device.Revision(r)
+	d.Address = obj.Attributes[Address]
+	d.Target = obj.Attributes[Target]
+	d.Version = obj.Attributes[Version]
+	t, _ := strconv.Atoi(obj.Attributes[Timeout])
```
Comment (SeanCondon) [inline api/topo/topo.go:78]: you should catch the error I think

File: api/topo/topo.go [hunk 3]
```diff
@@ -33,3 +58,86 @@ var TopoClientFactory = func(cc *grpc.ClientConn) TopoClient {
...
+
+	d := &device.Device{
+		ID:        device.ID(obj.ID),
+		Protocols: []*device.ProtocolState{},
+	}
+
+	r, _ := strconv.Atoi(obj.Attributes[Revision])
+	d.Revision = device.Revision(r)
+	d.Address = obj.Attributes[Address]
+	d.Target = obj.Attributes[Target]
```
Comment (SeanCondon) [inline api/topo/topo.go:76]: I think you should throw error if `address`, `type` or `version` are not found.

Will it be getting `type` from the Entity `kind`?

File: api/topo/topo.go [hunk 4]
```diff
@@ -33,3 +58,86 @@ var TopoClientFactory = func(cc *grpc.ClientConn) TopoClient {
...
+		Obj: &Object_Entity{
+			Entity: &Entity{
+				Protocols: []*device.ProtocolState{},
+			},
+		},
+	}
+	obj.Attributes[Revision] = strconv.Itoa(int(d.Revision))
+	obj.Attributes[Address] = d.Address
+	obj.Attributes[Target] = d.Target
+	obj.Attributes[Version] = d.Version
```
Comment (SeanCondon) [inline api/topo/topo.go:121]: missing `Type` - maybe it should be the `kind`

File: api/topo/topo.proto [hunk 5]
```diff
@@ -118,6 +119,7 @@ message Object {
 message Entity {
   // user-defined entity kind
   string kind_id = 1 [(gogoproto.customname) = "KindID", (gogoproto.casttype) = "ID"];
+  repeated topo.device.ProtocolState protocols = 2;
```
Comment (SeanCondon) [inline api/topo/topo.proto:122]: Maybe you should be moving this across in to the new hierarchy. I was wondering if we were going to bring this over in to the new hierarchy at all, but I think it is worthwhile. `onos-ric` also needs to update this protocol state like it is done here for `onos-config`

File: api/topo/topo.go (hunk 2 above)
Comment (shadansari) [reply api/topo/topo.go]: done

File: api/topo/topo.go (hunk 3 above)
Comment (shadansari) [reply api/topo/topo.go:73]: done

File: api/topo/topo.go (hunk 4 above)
Comment (shadansari) [reply api/topo/topo.go:121]: done

File: api/topo/topo.proto (hunk 5 above)
Comment (shadansari) [reply api/topo/topo.proto:122]: https://github.com/onosproject/onos-topo/issues/168

**Pull request merged**